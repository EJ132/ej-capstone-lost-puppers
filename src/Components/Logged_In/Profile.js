import React, { Component } from 'react'
import Navbar from '../Navbar'
import config from '../../config'
import '../../Stylesheets/Profile.css'
import TokenService from '../../services/token-service'
import DogTag from '../LostDogs/Dogtag'

export default class Profile extends Component {

    state = {
        user: {
            fullname: '',
            user_name: '',
            id: null
        },

        userPost : []
    }

    componentDidMount() {

        fetch(`${config.API_ENDPOINT}/profile/${TokenService.getUserName()}`, {
            headers: {'authorization': `bearer ${TokenService.getAuthToken()}`},})
        .then(res =>
            (!res.ok)
                ? res.json().then(e => Promise.reject(e))
                : res.json()
        )
        .then(resJSON => {
            this.setState({
                user: resJSON
            })
            TokenService.saveUserId(this.state.user.id)
            this.setCards()
        })

    }

    setCards = () => {

        return fetch(`${config.API_ENDPOINT}/pups`, {
            headers: {'authorization': `bearer ${TokenService.getAuthToken()}`},})
        .then(res =>
            (!res.ok)
                ? res.json().then(e => Promise.reject(e))
                : res.json()
        )
        .then(resJSON => {
            // eslint-disable-next-line eqeqeq
            const userPups = resJSON.filter(dog => {

                if(dog.owner === this.state.user.id){
                        return dog
                    }

                // eslint-disable-next-line array-callback-return
                return;
                })
            this.setState({
                userPost: userPups
            })
        })
    }

    
    render(){
        const {user, userPost} = this.state
        return (
            <div>
                <Navbar />
                <div className="profile_page">
                  <div className="profile">
                        <h2>WELCOME</h2>
                        <h3>{user.fullname}</h3>
                    </div>

                    <div className="Post">
                    <h2 id="yourPost">Your Posts</h2>
                        <div className="userPosts">
                        {userPost.map(dogCard => {
                            return <DogTag 
                                name={dogCard.name}
                                img={dogCard.image}
                                description={dogCard.description}
                                category={dogCard.category}
                                dateCreated={dogCard.date_created}
                                id={dogCard.id}
                                key={dogCard.id}
                                owner={dogCard.owner}
                                />
                        })}
                        </div>
                    </div>
                </div>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}