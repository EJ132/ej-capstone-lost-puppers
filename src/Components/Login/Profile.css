@import url('https://fonts.googleapis.com/css?family=Alegreya+Sans+SC:900|Asap&display=swap');

.profile_page {
    margin-top: 40px;
    min-height: 100vh;
    height: auto;
    background-color: white;
}

.profile {
    color: white;
    padding: 40px;
    background-color: #17BEBB;
    font-family: 'Alegreya Sans SC', sans-serif;
}

.profile h2 {
    font-size: 4rem;
    text-decoration: underline;
    letter-spacing: 1px;
}

.profile h3 {
    margin-top: 10px;
    font-size: 3rem;
    margin-left: 125px;
    letter-spacing: 2px;
}

.profile p {
    padding: 10px;
    margin-top: 10px;
}

.Post {
    font-family: 'Asap', sans-serif;
    height: auto;
    color: black;
    text-align: center;
}

#yourPost {
    font-size: 3rem;
    padding: 20px 10px 0px 10px;
}


.userPosts {
    width: 70%;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    padding: 50px;
    margin-left: auto;
    margin-right: auto;
}

.addPup {
    padding: 20px;
    margin-top: 15px
}

.addPup a {
    background-color: rgb(187, 177, 235);
    padding: 10px;
    display: block;
    width: 200px;
    margin-left: auto;
    margin-right: auto;
    border-radius: 5px;
    text-decoration: none;
    color: black;
}

@media screen and (max-width: 768px) {
    
    .profile {
        color: white;
        padding: 20px;
        height: auto;
        display: flex;
        flex-direction: column;
        width: 100%;
        text-align: center;
        margin-bottom: 0px;
    }
    
    .profile h2 {
        padding: initial;
        margin-top: 20px;
        margin-bottom: 10px;
    }
    
    .profile h3 {
        margin-left: initial;
    }
    
    .profile p {
        padding: 10px;
        margin-top: 10px;
    }
    
    .Post {
        height: auto;
        color: black;
        text-align: center;
    }
    
    
    .userPosts {
        width: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
        padding: 30px 50px 50px 50px;
        margin-left: auto;
        margin-right: auto;
    }
}