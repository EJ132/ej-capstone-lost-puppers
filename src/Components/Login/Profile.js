import React, { Component } from 'react'
import Navbar from '../NavBar/Navbar'
import './Profile.css'
import TokenService from '../../services/token-service'
import PupApiService from '../../services/thing-api-service'
import DogTag from '../LostDogs/Dogtag'
import config from '../../config'

export default class Profile extends Component {

    state = {
        user: {
            fullname: '',
            user_name: '',
            id: null
        },
        userPost : []
    }

    //FETCHES THE CARDS AND USER

    componentDidMount() {
        PupApiService.getProfile()
            .then(resJSON => {
                this.setState({
                    user: resJSON
                })
                TokenService.saveUserId(this.state.user.id)
                this.setCards()
            })
    }

    //FILTERS ONLY TO THE CARDS NEEDED(OWNED BY USER)

    setCards = () => {

        PupApiService.getpups()
        .then(resJSON => {
            // eslint-disable-next-line eqeqeq
            const userPups = resJSON.filter(dog => {

                if(dog.owner === this.state.user.id){
                        return dog
                    }

                // eslint-disable-next-line array-callback-return
                return;
                })
            this.setState({
                userPost: userPups
            })
        })
    }

    
    render(){
        const {user, userPost} = this.state
        return (
            <div>
                <Navbar />
                <div className="profile_page">
                  <div className="profile">
                        <h2>WELCOME</h2>
                        <h3>{user.fullname}</h3>
                    </div>

                    <div className="Post">
                    <h2 id="yourPost">Your Posts ({userPost.length})</h2>
                        <div className="userPosts">
                        {userPost.map(dogCard => {
                            return <DogTag 
                                name={dogCard.name}
                                img={dogCard.image}
                                description={dogCard.description}
                                category={dogCard.category}
                                dateCreated={dogCard.date_created}
                                id={dogCard.id}
                                key={dogCard.id}
                                owner={dogCard.owner}
                                />
                        })}
                        </div>
                    </div>
                </div>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}