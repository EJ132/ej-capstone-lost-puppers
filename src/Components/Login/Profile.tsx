import React, { Component } from "react";

import PupApiService from "../../services/thing-api-service";
import TokenService from "../../services/token-service";
import DogTag from "../LostDogs/Dogtag";
import NavigationBar from "../NavigationBar/NavigationBar";

import "./Profile.css";

export default class Profile extends Component {
  state = {
    user: {
      fullname: "",
      user_name: "",
      id: null,
    },
    userPost: [],
  };

  // FETCHES THE CARDS AND USER

  componentDidMount() {
    PupApiService.getProfile().then((resJSON) => {
      this.setState({
        user: resJSON,
      });
      TokenService.saveUserId(this.state.user.id);
      this.setCards();
    });
  }

  // FILTERS ONLY TO THE CARDS NEEDED(OWNED BY USER)

  setCards = () => {
    PupApiService.getpups().then((resJSON) => {
      // eslint-disable-next-line eqeqeq
      const userPups = resJSON.filter((dog) => {
        if (dog.owner === this.state.user.id) {
          return dog;
        }

        // eslint-disable-next-line array-callback-return
      });
      this.setState({
        userPost: userPups,
      });
    });
  };

  render() {
    const { user, userPost } = this.state;
    return (
      <div>
        <NavigationBar />
        <div className="profile_page">
          <div className="profile">
            <h2>WELCOME</h2>
            <h3>{user.fullname}</h3>
          </div>

          <div className="Post">
            <h2 id="yourPost">Your Posts ({userPost.length})</h2>
            <div className="userPosts">
              {userPost.map((dogCard) => {
                return (
                  <DogTag
                    name={dogCard.name}
                    img={dogCard.image}
                    description={dogCard.description}
                    category={dogCard.category}
                    dateCreated={dogCard.date_created}
                    id={dogCard.id}
                    key={dogCard.id}
                    owner={dogCard.owner}
                  />
                );
              })}
            </div>
          </div>
        </div>

        <footer>&#169; EJ Gonzalez 2019</footer>
      </div>
    );
  }
}
