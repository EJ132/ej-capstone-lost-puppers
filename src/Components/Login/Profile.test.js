import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Profile from './Profile'

describe(`Profile component`, () => {

  it('renders a Profile page by default', () => {
    const wrapper = shallow(<Profile />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
