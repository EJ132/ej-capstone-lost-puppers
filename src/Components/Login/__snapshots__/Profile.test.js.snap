// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders a Profile page by default 1`] = `
<div>
  <Navbar />
  <div
    className="profile_page"
  >
    <div
      className="profile"
    >
      <h2>
        WELCOME
      </h2>
      <h3 />
    </div>
    <div
      className="Post"
    >
      <h2
        id="yourPost"
      >
        Your Posts
      </h2>
      <div
        className="userPosts"
      />
    </div>
  </div>
  <footer>
    Â© EJ Gonzalez 2019
  </footer>
</div>
`;
